File: LeetCode-Java/Rotate List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode rotateRight(ListNode head, int n) {
        if(head == null) return null;
        int length = 1;
        ListNode current = head;
        while(current.next != null)
        {
            length ++;
            current = current.next;
        }
        n = n % length;
        if(n == 0) return head;
        ListNode test = head;
        for(int i = 0;i < length - n - 1;i ++)
        {
            test = test.next;
        }
        ListNode result = test.next;
        test.next = null;
        current.next=  head;
        return result;
    }
}

File: Leetcode-cpp/rotate_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if (head == nullptr) return head;
        ListNode* temp = head;
        int size = 0;
        while (temp != nullptr) {
            temp = temp->next;
            size++;
        }
        k = k % size;
        if (k == 0) return head;
        ListNode* slow = head;
        ListNode* fast = head;
        while (k > 0) {
            k--;
            fast = fast->next;
        }
        while (fast->next != nullptr) {
            fast = fast->next;
            slow = slow->next;
        }
        ListNode* result = slow->next;
        slow->next = nullptr;
        fast->next = head;
        return result;
    }
};